### PR TITLE
Include "assets/rust-toolchain" file to simplify hacspec workflow.

### PR DESCRIPTION
**Pull request by duesee**
This PR adds a `toolchain` file to the `assets` folder. The file is intended to be copied by people to the repository root who want to run the hacspec typechecker. (See README.md.) This simplifies working with Bertie.

If we don't add it, users will (likely) copy the file from the hacspec repository anyway or always pass `+<version>` (after looking it up from the hacspec repository). Having the file (and a few sentences in the README.md) here makes things more simple.

Alternative: Add it to .gitignore (root) and link to hacspec repo in README.md.

### CODE REVIEW

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
What's the purpose of this PR?

The nightly version in hacspec changes and always tracking that here is error prone. Also, it is not necessary to use the hacspec nightly version to build hacspec crates. It's only used for the hacspec typechecker.

**Comment by duesee** [comment]
I've updated the PR description.

File: README.md [hunk 1]
```diff
@@ -32,14 +32,22 @@ If you are looking for commercial support for Bertie, please [reach out to Cryps
 
 ### WORKING ON BERTIE
 
+Note: You do not need to do anything of this when you just want to build and run Bertie. This is only if you intend to *work* on Bertie, i.e., change its core and contribute the changes to the project.
```
Comment (franziskuskiefer) [inline README.md:37]:
```suggestion
Note: You do not need to do any of this when you just want to build and run Bertie. This is only if you intend to *work* on Bertie, i.e., change its core and contribute the changes to the project.
```

**Review by franziskuskiefer — APPROVED** [review]
lgtm with the nit

**Pull request merged**